code/obj/storage/gang_crate.dm

```diff
@@ -167,67 +172,193 @@
 /obj/item/gang_loot
 	icon = 'icons/obj/items/storage.dmi'
 	name = "suspicious looking duffle bag"
-	desc = "A greasy, black duffle bag, this isn't station issue..."
+	desc = "A greasy, black duffle bag, this isn't station issue, you should probably leave it alone..."
 	icon_state = "gang_dufflebag"
 	item_state = "bowling"
+	p_class = 4 //marginally easier than dragging a whole locker with this in
+	throw_range = 4
+	always_slow_pull = TRUE
 	w_class = W_CLASS_GIGANTIC
 	var/hidden = TRUE
 	var/open = FALSE
+	///Whether this bag's trap is active
+	var/trapped = TRUE
+	///Whether this bag is tracking its' location
+	var/tracking = TRUE
+	///Whether this bag is unopened. If TRUE, it will grant points when opened.
+	var/sealed = TRUE
+	///The gang who should own this duffel bag
+	var/datum/gang/owning_gang
+	///The name of the informant who knows about this bag
+	var/informant
+	///The civilian who has this in their hands, if the trap is active.
+	var/mob/living/idiot = null
+	///The area this bag spawned in.
+	var/area/start_area
+
+	///Items that haven't been removed from the bag. These will travel with it.
+	var/datum/vis_storage_controller/vis_controller = new()
+	var/datum/loot_generator/lootMaster
 	level = UNDERFLOOR
 
+
+	proc/initialize_loot_master(x,y)
+		src.vis_controller = new()
+		vis_controller.initialize(src)
+		lootMaster =  new /datum/loot_generator(x,y)
+		RegisterSignal(src, XSIG_MOVABLE_AREA_CHANGED, PROC_REF(alert_gang))
```

Comment:
`toggle_tracking()` should handle all signals related to tracking; see my first review comment for initialising tracking.